### PR TITLE
TD3

### PR DESCRIPTION
**Pull request by gal-leibovich**
Ready for review - reproducing and even surpassing paper results.

Note: Slight renaming of the `noise_percentage_schedule` parameter resulted in some changes in unrelated files. 

Also, some small fixes to DDPG.

### CODE REVIEW

File: rl_coach/agents/ddpg_agent.py [hunk 1]
```diff
@@ -23,11 +23,12 @@
 from rl_coach.agents.actor_critic_agent import ActorCriticAgent
```
Comment (gal-leibovich) [inline rl_coach/agents/ddpg_agent.py]: Ignore changes in this file, it will be removed before merging. Only leaving here so to track the changes on top of DDPG.

File: rl_coach/agents/td3_agent.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            else:
+                actor.apply_gradients_to_online_network(gradients)
+
+        return total_loss, losses, unclipped_grads
+
+    def train(self):
+        self.ap.algorithm.num_consecutive_training_steps = self.current_episode_steps_counter
+        return Agent.train(self)
+
+    def choose_action(self, curr_state):
```
Comment (gal-leibovich) [inline rl_coach/agents/td3_agent.py]: TODO - inherit this method from DDPG

**Comment by gal-leibovich** [comment]
Comparison of TD3 (called FixedCriticInputBugAndBootstrapBug) to DDPG (no batchnorm)

The TD3 implementation in this repo significantly outperforms the paper published results.

**Hopper**
image

**Walker2D**
image

**Reacher**
image

**Half Cheetah**
image

**Ant**
image

File: rl_coach/architectures/tensorflow_components/heads/td3_v_head.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        self.loss_type = []
+        self.initializer = initializer
+        self.loss = []
+        self.output = []
+
+    def _build_module(self, input_layer):
+        # Standard V Network
+        q_outputs = []
+        self.target = tf.placeholder(tf.float32, shape=(None, 1), name="q_networks_min_placeholder")
+
```
Comment (galnov) [inline rl_coach/architectures/tensorflow_components/heads/td3_v_head.py]: Worth adding a comment stating that the TD agent assumes that input_layer.shape[0] is 2.

File: rl_coach/exploration_policies/additive_noise.py [hunk 4]
```diff
@@ -48,17 +48,17 @@ class AdditiveNoise(ContinuousActionExplorationPolicy):
...
+    def __init__(self, action_space: ActionSpace, noise_schedule: Schedule,
+                 evaluation_noise: float, noise_as_percentage_from_action_space: bool = True):
         """
         :param action_space: the action space used by the environment
-        :param noise_percentage_schedule: the schedule for the noise variance percentage relative to the absolute range
-                                          of the action space
-        :param evaluation_noise_percentage: the noise variance percentage that will be used during evaluation phases
+        :param noise_schedule: the schedule for the noise
+        :param evaluation_noise: the noise variance that will be used during evaluation phases
         """
```
Comment (galnov) [inline rl_coach/exploration_policies/additive_noise.py]: Missing documentation of noise_as_percentage_from_action_space
